text/0049-match-arm-attributes.md

```diff
@@ -0,0 +1,82 @@
+- Start Date: 2014-03-20
+- RFC PR: [rust-lang/rfcs#49](https://github.com/rust-lang/rfcs/pull/49)
+- Rust Issue: [rust-lang/rust#12812](https://github.com/rust-lang/rust/issues/12812)
+- Translators: [[@FizzyElt](https://github.com/FizzyElt)]
+- Commit: [The commit link this page based on](https://github.com/rust-lang/rfcs/blob/master/text/0049-match-arm-attributes.md)
+- Updated: 2022-05-17
+
+# 概要
+
+允許屬性在匹配分支上
+
+# 動機
+
+有時希望使用屬性來注釋匹配語句的分支，例如條件編譯 `#[cfg]` 或是分支權重（後者為最重要的用途）。
+
+
+對於條件編譯，解決辦法是使用 `#[cfg]` 來重複包含函式的全部部分。一個案例研究是 [sfackler's bindings to OpenSSL](https://github.com/sfackler/rust-openssl)，在多數的發行版本中移除了 SSLv2 支持，因此 Rust bindings 部分需要被條件禁用。支持各種不同 SSL 版本的明顯方法是一個枚舉。
+
+```rust
+pub enum SslMethod {
+    #[cfg(sslv2)]
+    /// Only support the SSLv2 protocol
+    Sslv2,
+    /// Only support the SSLv3 protocol
+    Sslv3,
+    /// Only support the TLSv1 protocol
+    Tlsv1,
+    /// Support the SSLv2, SSLv3 and TLSv1 protocols
+    Sslv23,
+}
+```
+
+然而，所有的 `match` 只能在 `cfg` 是有效的時候提起 `Sslv2`，例如下面內容是無效的：
+
+```rust
+fn name(method: SslMethod) -> &'static str {
+    match method {
+        Sslv2 => "SSLv2",
+        Sslv3 => "SSLv3",
+        _ => "..."
+    }
+}
+```
+
+一個有效的方法有兩個定義：`#[cfg(sslv2)] fn
+name(...)` 和 `#[cfg(not(sslv2)] fn name(...)`
```

Comment:
I wonder if this will match the original sentence more 👀 ?

```suggestion
一個有效的方法則必須有兩個定義：`#[cfg(sslv2)] fn
name(...)` 和 `#[cfg(not(sslv2)] fn name(...)`
```